### ISSUE TITLE
[Bug] CarouselView disappear in iOS

### ISSUE THREAD

**Issue by AbuMandour** [xamarin/Xamarin.Forms#13255]
Carouselview disappear in iOS when setting HeightRequest and PeekAreaInsets
### Description
when I set height request less than 190 and set PeekAreaInsets to any value it disappears from iOS
### Steps to Reproduce

1. create Carosuelview
2. set HeightRequest less than 200
3. set PeekAreaInset

### Expected Behavior
![Simulator Screen Shot - iPhone 11 - 2020-12-30 at 07 01 59](https://user-images.githubusercontent.com/23154124/103331240-9d8baf00-4a6d-11eb-8222-9916465890b2.png)

### Actual Behavior
![Simulator Screen Shot - iPhone 11 - 2020-12-30 at 07 02 08](https://user-images.githubusercontent.com/23154124/103331245-a41a2680-4a6d-11eb-88fc-3e0a5fa4f9e1.png)

### Basic Information

- Version with issue: Xamarin.Forms 5.0.0.1829-pre6
- Last known good version:
- Platform Target Frameworks: 
  - iOS:  Xamarin.Forms 5.0.0.1829-pre6
  - Android: Xamarin.Forms 5.0.0.1829-pre6
- Android Support Library / AndroidX Version:
- NuGet Packages: 5.8.0.6860
- Affected Devices: iOS simulator and Iphone

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.1 (build 37)
Installation UUID: c069a490-fad2-44c6-8c2a-efc77bdf173b
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000093

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.93 (2020-02/620cf538206) (64-bit)
	Package version: 612000093

=== Roslyn (Language Service) ===

3.8.0-5.20515.7+713efb0e68d34ab75faaf709e071e5c2d8d4ed0e

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.100/Sdks
SDK Versions:
	5.0.100
	3.1.403
	3.1.402
	3.1.300
	3.1.201
	3.1.200
	3.1.101
	3.0.103
	3.0.101
	3.0.100
	3.0.100-preview9-014004
	2.2.401
	2.1.701
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.0
	3.1.9
	3.1.8
	3.1.4
	3.1.3
	3.0.3
	2.1.23
	2.1.22
	2.1.18
	2.1.16

=== .NET Core 3.1 SDK ===

SDK: 3.1.403

=== Xamarin.Profiler ===

Version: 1.6.13.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 12.2 (17535)
Build 12B45b

=== Xamarin.Mac ===

Version: 7.0.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:40-0500

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin Designer ===

Version: 16.8.0.507
Hash: e87b24884
Branch: remotes/origin/d16-8
Build date: 2020-10-29 00:31:38 UTC

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Community)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/abumandour/Library/Android/android-sdk-macosx
	Supported Android versions:
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/abumandour/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:30 UTC

=== Android Device Manager ===

Version: 16.8.0.45
Hash: fc0af5f
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:54 UTC

=== Build Information ===

Release ID: 808010037
Git revision: acd8ddc086768c701bd641d6ec27316f45813175
Build date: 2020-11-17 12:49:41-05
Build branch: release-8.8
Xamarin extensions: acd8ddc086768c701bd641d6ec27316f45813175

=== Operating System ===

Mac OS X 10.15.7
Darwin 19.6.0 Darwin Kernel Version 19.6.0
    Mon Aug 31 22:12:52 PDT 2020
    root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

Essential UI Kit for Xamarin.Forms - Syncfusion 3.0.0.0
XAML Styler 2.0.1


```

### Reproduction Link
this code works fine
`
        <CarouselView ItemsSource="{Binding Monkeys}"
                      HeightRequest="200"
                      PeekAreaInsets="100">
`
this code doesn't work
`
        <CarouselView ItemsSource="{Binding Monkeys}"
                      HeightRequest="150"
                      PeekAreaInsets="100">
`

**Comment by rachelkang** [comment]
Hi, @AbuMandour - thanks for submitting this issue. I'm having some trouble reproducing the behavior you've shared. Based on the screenshots, I'm assuming you've started from the CarouselViewDemos VisualStatePage - would you mind sharing what else you've added to the sample code? And what emulator/device are you testing on?

For some reason, changing the `HeightRequest` as you've indicated is not changing anything for me:
![Screenshot 2021-01-06 at 1 30 05 PM](https://user-images.githubusercontent.com/21988533/103806887-a6f3b900-5023-11eb-8d5d-a0ce4ff89ee2.png)

**Comment by AbuMandour** [comment]
@rachelkang Hello Rachel, I think that change height in your code won't change anything as you don't wrap carousel into any parent layout so it should take all height of the page if you wrap it into stack layout which will take all page height then set the height of carousel view you will see the issue.
in the other hand, I reach what is the issue it seems that if I set PeekAreaInsets in android it respect the orientation of carousel view so if I set  `PeekAreaInsets="70"` it will convert it implicated to `PeekAreaInsets="70,0"` but it wouldn't do the same in iOS so when I set it explicated `PeekAreaInsets="70,0"` it works fine.

**Comment by rachelkang** [comment]
@AbuMandour ah, ok yes - I wrapped it in a StackLayout and see what you described now!
@rmarinho do you know what might be happening here? When the HeightRequest of CarouselView is set low (i.e. 150), it seems to disappear entirely

As for the other issue you described, however, I'm not seeing that behavior. It seems to work fine, but if you're still having issues and can share more / potentially file a separate bug report, that would be great!
![Screenshot 2021-01-08 at 1 27 50 PM](https://user-images.githubusercontent.com/21988533/104050991-6deb4e00-51b5-11eb-9923-f27f2021e6b0.png)

**Comment by jameswestgate** [comment]
I'm also seeing a similar issue when moving from Forms 4.8.0.1821 to 5.0. In my case, the carousel is inside a ScrollView.

**Comment by Redth** [comment]
Yes could you please attach a sample which reproduces the issue?

**Comment by jameswestgate** [comment]
Hi @Redth , 

I have reduced this to a very simple example here: https://github.com/jameswestgate/CarouselRegression
The issue seems to be with ContentView and not ScrollView as previously suggested.
Can confirm this issue still exists on build 5.0.0.1931 and works as soon as I revert to 4.8.0.1821

Many thanks!

**Comment by jameswestgate** [comment]
Hi @rachelkang 

Please see #13636. My issue was caused by the new `Loop` property being set to true by default and then breaking rendering. I can disable looping to work around the problem at this time.
